app/services/sign_in/session_revoker.rb

```diff
@@ -78,5 +80,13 @@ def delete_session!
     ensure
       session.destroy!
     end
+
+    def delete_device_sessions
+      hashed_device_secret = get_hash(device_secret)
+      sessions = OAuthSession.where(hashed_device_secret:)
```

Comment:
You can do `OAuthSession.where(hashed_device_secret:).destroy_all` to do this a bit more succinctly